test/parallel/test-stream-writable-end-cb-error.js

```diff
@@ -46,3 +46,30 @@ const stream = require('stream');
     writable.emit('error', new Error('kaboom'));
   }));
 }
+
+{
+  const w = new stream.Writable({
+    write(chunk, encoding, callback) {
+      setImmediate(callback);
+    },
+    finish(callback) {
+      setImmediate(callback);
+    }
+  });
+  w.end('testing ended state', common.mustCall());
+  assert.strictEqual(w.destroyed, false);
+  assert.strictEqual(w.writableEnded, true);
+  w.end(common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_STREAM_WRITE_AFTER_END');
+  }));
+  assert.strictEqual(w.destroyed, false);
+  assert.strictEqual(w.writableEnded, true);
+  w.end('end', common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_STREAM_WRITE_AFTER_END');
+  }));
+  assert.strict(w.destroyed, true);
```

Comment:
?

Reply:
?

Reply:
`assert.strictEqual()`?

Reply:
Because `.end(data)` causes error which call `stream.destroy(err)`.

Reply:
Can you remind me why we make a distinction between `.end()` and `.end(data)`?

Reply:
Calling `end()` doesn't lose data.

Reply:
hmm ok.